pkg/splunk/enterprise/util.go

```diff
@@ -1928,3 +1928,88 @@ func migrateAfwFromPhase2ToPhase3(ctx context.Context, client splcommon.Controll
 func isAppFrameworkMigrationNeeded(afwStatusContext *enterpriseApi.AppDeploymentContext) bool {
 	return afwStatusContext != nil && afwStatusContext.Version < currentAfwVersion && len(afwStatusContext.AppsSrcDeployStatus) > 0
 }
+
+// updateCRStatus fetches the latest CR, and on top of that, updates latest status
+func updateCRStatus(ctx context.Context, client splcommon.ControllerClient, dirtyCR splcommon.MetaObject) {
+	reqLogger := log.FromContext(ctx)
+	scopedLog := reqLogger.WithName("updateCRStatus").WithValues("name", dirtyCR.GetName(), "namespace", dirtyCR.GetNamespace())
+
+	//sgontla: Fix the UT to accommodate multiple retries
+	for i := 0; i < 1; i++ {
+		cr, err := fetchCurrentCRWithStatusUpdate(ctx, client, dirtyCR)
+		if err != nil {
+			scopedLog.Error(err, "Unable to Read latest CR from the K8s")
+			continue
+		}
+
+		err = client.Status().Update(ctx, cr)
+		if err == nil {
+			scopedLog.Info("Status update successful", "cr version", cr.GetResourceVersion(), "Try count", i)
+			break
+		}
+	}
```

Comment:
we need time.Sleep(10 * time.Millisecond) when update fails before trying for the second time